plugins/outputs/logzio/logzio.go

```diff
@@ -0,0 +1,185 @@
+package logzio
+
+import (
+	. "bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+const (
+	defaultLogzioRequestTimeout = time.Second * 5
+	defaultLogzioURL            = "https://listener.logz.io:8071"
+
+	logzioDescription        = "Send aggregate metrics to Logz.io"
+	logzioType               = "telegraf"
+	logzioMaxRequestBodySize = 9 * 1024 * 1024 // 9MB
+)
+
+var sampleConfig = `
+  ## Logz.io account token
+  token = "your logz.io token" # required
+
+  ## Use your listener URL for your Logz.io account region.
+  # url = "https://listener.logz.io:8071"
+
+  ## Timeout for HTTP requests
+  # timeout = "5s"
+
+  ## Optional TLS Config for use on HTTP connections
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+type Logzio struct {
+	Token   string            `toml:"token"`
+	URL     string            `toml:"url"`
+	Timeout internal.Duration `toml:"timeout"`
+	tls.ClientConfig
+
+	client *http.Client
+}
+
+// Connect to the Output
+func (l *Logzio) Connect() error {
+	log.Printf("D! [outputs.logzio] Connecting to logz.io output...\n")
+	if l.Token == "" || l.Token == "your logz.io token" {
+		return fmt.Errorf("token is required")
+	}
+
+	if l.URL == "" {
+		l.URL = defaultLogzioURL
+	}
+
+	if l.Timeout.Duration == 0 {
```

Comment:
You would like to probably also check for negative values (`internal.Duration` allows them). For example: `if l.Timeout.Duration <= 0`